### PR TITLE
refactor: Cleanup of events, reducers and actions 

### PR DESCRIPTION
**Pull request by gregor**

### CODE REVIEW

File: electron/js/lib/googleAuth.js [hunk 1]
```diff
@@ -41,12 +40,13 @@ const _authorizeApp = (url) => {
       setImmediate(() => {
         const title = win.getTitle();
 
+        const [, , returnValue] = title.split(/[ =]/);
```
Comment (ffflorian) [inline electron/js/lib/googleAuth.js]:
```js
const returnValue = title.split(/[ =]/)[2];
```

File: electron/js/lib/openGraph.js [hunk 2]
```diff
@@ -41,13 +41,13 @@ const _fetchImageAsBase64 = (url) => {
   });
 };
 
-const _fetchOpenGraphData = (url) => {
+const fetchOpenGraphData = (url) => {
```
Comment (ffflorian) [inline electron/js/lib/openGraph.js]:
```js
const fetchOpenGraphData = url => {
```

File: electron/js/lib/openGraph.js [hunk 3]
```diff
@@ -57,14 +57,14 @@ const _updateMetaDataWithImage = (meta, image) => {
   return meta;
 };
 
-const _getOpenGraphData = (url, callback) => {
-  return _fetchOpenGraphData(url)
+const getOpenGraphData = (url, callback) => {
+  return fetchOpenGraphData(url)
     .then((meta) => {
```
Comment (ffflorian) [inline electron/js/lib/openGraph.js]:
```js
.then(meta => {
```

File: electron/js/lib/openGraph.js [hunk 4]
```diff
@@ -57,14 +57,14 @@ const _updateMetaDataWithImage = (meta, image) => {
...
       if (meta.image && meta.image.url) {
-        const [imageUrl] = _arrayify(meta.image.url);
+        const [imageUrl] = arrayify(meta.image.url);
 
-        return _fetchImageAsBase64(imageUrl)
-          .then((uri) => _updateMetaDataWithImage(meta, uri))
+        return fetchImageAsBase64(imageUrl)
+          .then((uri) => updateMetaDataWithImage(meta, uri))
```
Comment (ffflorian) [inline electron/js/lib/openGraph.js]:
```js
.then(uri => updateMetaDataWithImage(meta, uri))
```

File: electron/js/certutils.js [hunk 5]
```diff
@@ -59,11 +59,9 @@ const pins = [
 ];
 
 module.exports = {
-  hostnameShouldBePinned (hostname) {
-    return pins.some(pin => pin.url.test(hostname.toLowerCase().trim()));
-  },
+  hostnameShouldBePinned: hostname => pins.some(pin => pin.url.test(hostname.toLowerCase().trim())),
```
Comment (atomrc) [inline electron/js/certutils.js]: Is there a real benefit to this?

File: electron/renderer/src/actions/index.js [hunk 6]
```diff
@@ -16,54 +16,58 @@ export const addAccount = (withSession = true) => {
...
 
-export const deleteAccount = id => {
+export const updateAccountBadge = (id, count) => {
   return {
+    count,
     id,
-    type: DELETE_ACCOUNT,
+    type: UPDATE_ACCOUNT_BADGE,
   };
 };
```
Comment (ffflorian) [inline electron/renderer/src/actions/index.js]:
```js
export const updateAccountBadge = (id, count) => ({
  count,
  id,
  type: UPDATE_ACCOUNT_BADGE,
});
```

Same for the other functions in this file.

File: electron/renderer/src/actions/index.js [hunk 7]
```diff
@@ -16,54 +16,58 @@ export const addAccount = (withSession = true) => {
...
 
 export const toggleAddAccountMenuVisibility = (x, y) => {
   return {
-    payload: { position: { x, y } },
-    type: 'TOGGLE_ADD_ACCOUNT_VISIBILITY',
+    payload: {position: {x, y}},
+    type: TOGGLE_ADD_ACCOUNT_VISIBILITY,
   };
 };
```
Comment (ffflorian) [inline electron/renderer/src/actions/index.js]:
```js
export const toggleAddAccountMenuVisibility = (x, y) => ({
  payload: {position: {x, y}},
  type: TOGGLE_ADD_ACCOUNT_VISIBILITY,
});
```

File: electron/renderer/src/actions/index.js [hunk 8]
```diff
@@ -16,54 +16,58 @@ export const addAccount = (withSession = true) => {
...
+export const HIDE_CONTEXT_MENUS = 'HIDE_CONTEXT_MENUS';
+export const TOGGLE_ADD_ACCOUNT_VISIBILITY = 'TOGGLE_ADD_ACCOUNT_VISIBILITY';
+export const TOGGLE_EDIT_ACCOUNT_VISIBILITY = 'TOGGLE_EDIT_ACCOUNT_VISIBILITY';
+
 export const setAccountContextHidden = () => {
   return {
-    type: 'HIDE_CONTEXT_MENUS',
+    type: HIDE_CONTEXT_MENUS,
   };
 };
```
Comment (ffflorian) [inline electron/renderer/src/actions/index.js]:
```js
export const setAccountContextHidden = () => ({
  type: HIDE_CONTEXT_MENUS,
});
```

File: electron/renderer/src/components/Webview.js [hunk 9]
```diff
@@ -58,15 +58,15 @@ class Webview extends Component {
   }
 
   _focusWebview() {
-    const is_visible = this.props.visible === true;
-    if (is_visible) {
+    const isVisible = this.props.visible === true;
+    if (isVisible) {
```
Comment (ffflorian) [inline electron/renderer/src/components/Webview.js]:
```js
if (this.props.visible === true) {
```

File: electron/js/certutils.js (hunk 5 above)
Comment (lipis) [reply electron/js/certutils.js]: the original version is more readable I would say..

File: electron/renderer/src/lib/eventType.js [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+ *
+ */
+
+export const ACCOUNT = {
+  UPDATE_INFO: 'EVENT_TYPE.ACCOUNT.UPDATE_INFO',
+};
+export const ACTION = {
+  NOTIFICATION_CLICK: 'EVENT_TYPE.ACTION.NOTIFICATION_CLICK',
+};
+export const LIFECYCLE = {
```
Comment (ffflorian) [inline electron/renderer/src/lib/eventType.js]:
```js
};

export const LIFECYCLE = {
```

File: electron/renderer/src/lib/localStorage.js [hunk 11]
```diff
@@ -33,7 +32,7 @@ export const loadState = () => {
 export const saveState = (state) => {
```
Comment (ffflorian) [inline electron/renderer/src/lib/localStorage.js]:
```js
export const saveState = state => {
```

File: electron/js/lib/googleAuth.js (hunk 1 above)
Comment (AndyLnd) [inline electron/js/lib/googleAuth.js]: is this really better than ```const returnValue = title.split(/[ =]/)[2];```?

File: electron/renderer/src/reducers/accountReducer.js [hunk 12]
```diff
@@ -18,59 +18,66 @@
...
     name: undefined,
     picture: undefined,
-    sessionID: sessionID,
+    sessionID: sessionId,
     teamID: undefined,
     userID: undefined,
     visible: true,
   };
-}
+};
```
Comment (ffflorian) [inline electron/renderer/src/reducers/accountReducer.js]:
```js
const createAccount = sessionId => ({
  accentID: undefined,
  badgeCount: 0,
  id: uuid(),
  lifecycle: undefined,
  name: undefined,
  picture: undefined,
  sessionID: sessionId,
  teamID: undefined,
  userID: undefined,
  visible: true,
});

File: electron/renderer/src/lib/localStorage.js (hunk 11 above)
Comment (lipis) [reply electron/renderer/src/lib/localStorage.js]: these things will be fixed automagically once the #1058 will be fixed and eventually Prettier (#1035)

File: electron/renderer/src/components/Webview.js (hunk 9 above)
Comment (lipis) [reply electron/renderer/src/components/Webview.js]: or just `if (this.props.visible)`?

File: electron/renderer/src/lib/localStorage.js (hunk 11 above)
Comment (ffflorian) [reply electron/renderer/src/lib/localStorage.js]: Well, yes, eventually...

File: electron/renderer/src/components/Webview.js (hunk 9 above)
Comment (ffflorian) [reply electron/renderer/src/components/Webview.js]: I prefer the explicit test.

File: electron/js/certutils.js (hunk 5 above)
Comment (ffflorian) [reply electron/js/certutils.js]: It's shorter :smiley:

File: electron/renderer/src/lib/localStorage.js (hunk 11 above)
Comment (lipis) [reply electron/renderer/src/lib/localStorage.js]: Maybe even tomorrow :) as @gregor has some ideas :) Robots will do their job, eventually..

File: electron/renderer/src/reducers/accountReducer.js [hunk 13]
```diff
@@ -18,59 +18,66 @@
...
 
-const accounts = (state = [createAccount()], action) => {
+const accountReducer = (state = [createAccount()], action) => {
```
Comment (Yserz) [inline electron/renderer/src/reducers/accountReducer.js]: 👍

File: electron/js/lib/googleAuth.js (hunk 1 above)
Comment (gregor) [reply electron/js/lib/googleAuth.js]: But it fails as soon as you enable magic number checking

File: electron/renderer/static/webview-preload.js [hunk 14]
```diff
@@ -156,12 +152,7 @@ const enableFileLogging = () => {
   const id = new URL(window.location).searchParams.get('id');
 
   if (id) {
-    const logFilePath = path.join(
-      app.getPath('userData'),
-      'logs',
-      id,
-      config.LOG_FILE_NAME
-    );
+    const logFilePath = path.join(app.getPath('userData'), 'logs', id, config.LOG_FILE_NAME);
```
[the remark this answers is not in the document]
Comment (gregor) [reply electron/renderer/static/webview-preload.js]: Like the idea but I don't want to add more stuff to this PR than we already have.

File: electron/renderer/static/webview-preload.js [hunk 15]
```diff
@@ -197,29 +185,20 @@ const checkAvailability = callback => {
...
   global.setImmediate = _setImmediate;
   global.desktopCapturer = desktopCapturer;
   global.environment = environment;
   global.openGraph = require('../../js/lib/openGraph');
-  global.notification_icon = path.join(
-    app.getAppPath(),
-    'img',
-    'notification.png'
-  );
+  global.notification_icon = path.join(app.getAppPath(), 'img', 'notification.png');
```
[the remark this answers is not in the document]
Comment (gregor) [reply electron/renderer/static/webview-preload.js]: See above

File: electron/renderer/src/components/Webview.js (hunk 9 above)
Comment (lipis) [reply electron/renderer/src/components/Webview.js]: plus [it's shorter](https://github.com/wireapp/wire-desktop/pull/1382#discussion_r184029420) :)

**Pull request merged**